src/features/surveys/panes/SurveySubmissionPane.tsx

```diff
@@ -57,10 +58,14 @@ const useStyles = makeStyles({
   },
 });
 
-const SurveySubmissionPane: FC<SurveySubmissionPaneProps> = ({ orgId, id }) => {
+const SurveySubmissionPane: FC<SurveySubmissionPaneProps> = ({
+  orgId,
+  id,
+  surveyId,
```

Comment:
This should not have to require the surveyId. I understand why it does with your refactor, but the surveyId is not necessary to load the submission from the API, only to find it in our cache, so that is a bit of a shortcoming of the cache.

I can think of two solutions:

1. Either refactor the store so that it includes _both_ a flat list of submissions, and a list by survey ID. This is how we do it in multiple other stores. Check out the events store for example, which contains both `eventList` and multiple `eventsBy*` maps.
2. Or change `useSurveySubmission` to look in all the maps to find if the submission has already been loaded. Less efficient, but also less duplication of data so less error-prone probably. Will elaborate in comment on `useSurveySubmission`.